core/dbt/tests/util.py

```diff
@@ -69,7 +71,10 @@ def run_dbt(args: List[str] = None, expect_pass=True):
 
     print("\n\nInvoking dbt with {}".format(args))
     res, success = handle_and_check(args)
-    assert success == expect_pass, "dbt exit state did not match expected"
+
+    if expect_pass is not None:
```

Comment:
I don't see the reason for this change. "expect_pass" should always be True or False.

Reply:
If `expect_pass` is a binary option, then the exit code of the `dbt <subcommand> <options>` will preempt everything else and take priority over subsequent assertions. Having an option to `run_dbt` that doesn't `assert` allows the user to take control of the assertion process independent of the exit code.

Why this matters in this case:
- `dbt debug` will have a non-zero exit code in certain situations (like if a profile is not found at a particular location).

In these tests, I want to get the output of `dbt debug` _regardless_ of the exit code and test the content.

Reply:
### Overview
`pytest` recommends thinking of tests being composed of [four steps](https://docs.pytest.org/en/latest/explanation/anatomy.html#test-anatomy):
1. Arrange
1. Act
1. Assert
1. Cleanup

Currently, the implementation of `dbt_run` combines both "Act" and "Assert". I want the _option_ for it to only "Act" and leave the "Assert" for a separate and independent step. This also seems like it would align with the anatomy of a test that `pytest` is promoting.

### Why
This pull request is a prelude to testing any implementation of https://github.com/dbt-labs/dbt-core/issues/5411.

As-is, `expect_pass=False` _might_ work for the ~9 different `dbt_run`s that end up being executed. But an implementation of #5411 will introduce another ~3 executions of `dbt_run`s (actually doing `dbt debug`) which might actually have a zero exit codes, hence those specific cases would need to be `expect_pass=True`.

It's greatly simplified if there is the option to separate "Assert" from "Act" within `dbt_run`.

Additionally, this would give more flexibility to other test implementers (while retaining the current default value of `expect_pass=True`).

Are there downsides or risks you see in my proposed change?

Reply:
I don't really object to this change. It mostly works this way because that's the way it was in test/integration/base.py and it's just convenient to not have to do an assert for almost every dbt_run.